### PR TITLE
Plot tracks in latent space and real space

### PR DESCRIPTION
**Pull request by mattersoflight**
This PR aims to add plotting utilities to visualize features (or their dimensionally reduced versions) over tracks of single cells, alongside images of the same tracked cell. 

- [x] Implement visualization of single track: all features (X-dimension: feature, Y-dimension: time), UMAP or PCA (track in UMAP1-UMAP2 space or PCA1-PCA2 space), images of the same cell. 
- [ ] overlay visualization of multiple tracks

### CODE REVIEW

File: applications/contrastive_phenotyping/contrastive_cli/plot_embeddings.py [hunk 1]
```diff
@@ -16,6 +16,13 @@
 )
 dataset
 
+# %%
+# Extract a track from the dataset
+all_tracks_FOV = dataset.sel(fov_name=slice("A/4/0"))
+a_track_in_FOV = all_tracks_FOV.sel(track_id=slice(23))
+print(a_track_in_FOV.sizes())
+# Why is sample dimension ~22000 long after the dataset is sliced by FOV and by track_id?
```
Comment (mattersoflight) [inline applications/contrastive_phenotyping/contrastive_cli/plot_embeddings.py:24]: @ziw-liu how can I slice the dataset by specified fov_name and specific track_id? The above lines of code don't return the dataset of expected dimensions. I would expect a dataset with ~10-30 samples, but get ~22000 samples.

**Review by mattersoflight — COMMENTED** [review]
@ziw-liu let's use this PR to coordinate development of the plotting features we need next.

File: applications/contrastive_phenotyping/contrastive_cli/plot_embeddings.py (hunk 1 above)
Comment (ziw-liu) [reply applications/contrastive_phenotyping/contrastive_cli/plot_embeddings.py:24]: A sample is one pair of feature/projected vector (per node `id` in tracking graph). One `track_id` contains many `id`s.

Comment (ziw-liu) [reply applications/contrastive_phenotyping/contrastive_cli/plot_embeddings.py:24]: And here by using `slice(23)` you are slicing multiple `track_id`s:

```py
assert len(list(range(26))[slice(23)]) == 23
```

Comment (mattersoflight) [reply applications/contrastive_phenotyping/contrastive_cli/plot_embeddings.py:24]: Your explanation is not clear to me.
Here is the logic I am following:
* I slice all tracks in FOV "A/4/0" and go from 6500 samples (65000 pairs features/projections) to ~500 samples (track_ids x time points). 
* I slice a track_id in the above dataset to get ~30 samples for specific track_ids.

>  One track_id contains many ids.
In other words, I am trying to index samples by fov_name and track_id

I am not familiar with xarray indexing API. Please implement the above and push your change.

Comment (ziw-liu) [reply applications/contrastive_phenotyping/contrastive_cli/plot_embeddings.py:24]: This gives 38 samples:
```suggestion
# %%
# Extract a track from the dataset
all_tracks_FOV = dataset.sel(fov_name="/A/4/0")
a_track_in_FOV = all_tracks_FOV.sel(track_id=23)
```

Comment (mattersoflight) [reply applications/contrastive_phenotyping/contrastive_cli/plot_embeddings.py:24]: >  asert len(list(range(26))[slice(23)]) == 23
I see that I am slicing indices 1:23. and "A/3/0":"A/4/0". Can't figure out how to index using only those coordinate values.

Comment (mattersoflight) [reply applications/contrastive_phenotyping/contrastive_cli/plot_embeddings.py]: thanks @ziw-liu

File: applications/contrastive_phenotyping/contrastive_cli/plot_embeddings.py [hunk 2]
```diff
@@ -16,6 +16,43 @@
 )
```
Comment (ziw-liu) [inline applications/contrastive_phenotyping/contrastive_cli/plot_embeddings.py:16]: I changed the path to:

```sh
/hpc/projects/intracellular_dashboard/viral-sensor/infection_classification/models/contrastive_tune_augmentations/predict/2024_02_04/tokenized-drop_path_0_0.zarr
```

**Comment by mattersoflight** [comment]
Current [commit](https://github.com/mehta-lab/VisCy/pull/135/commits/1debe64cd8d5431425dd21c8c1bb033e6f576801) shows how to use TripletDataModule to load data from a specific track_id and render in napari

**Comment by ziw-liu** [comment]
@Soorya19Pradeep Note that the animated plot will be missing legend items (e.g. infected) if the class is not present in the first frame. This is an [upstream issue](https://github.com/plotly/plotly.js/issues/2861).

**Comment by mattersoflight** [comment]
@Soorya19Pradeep as discussed offline, please refactor this script to generate a "report of embeddings" with a curated test set of cell divisions and infections. Let's compare these plots (on confluence) for the model you just trained and the similar model trained with a larger FOV.

**Pull request merged**